internal/provider/schemas/team.go

```diff
@@ -67,3 +74,81 @@ func TeamDataSourceSchemaAttributes() map[string]datasourceSchema.Attribute {
 		},
 	}
 }
+
+func TeamResourceSchemaAttributes() map[string]resourceSchema.Attribute {
+	return map[string]resourceSchema.Attribute{
+		"id": resourceSchema.StringAttribute{
+			MarkdownDescription: "Team identifier",
+			Computed:            true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"name": resourceSchema.StringAttribute{
+			MarkdownDescription: "Team name",
+			Required:            true,
+		},
+		"description": resourceSchema.StringAttribute{
+			MarkdownDescription: "Team description",
+			Optional:            true,
+		},
+		"member_ids": resourceSchema.SetAttribute{
+			ElementType:         types.StringType,
+			MarkdownDescription: "The IDs of the users to add to the Team",
+			Optional:            true,
+			Validators: []validator.Set{
+				setvalidator.SizeAtLeast(1),
+			},
+		},
+		"is_idp_managed": resourceSchema.BoolAttribute{
+			MarkdownDescription: "Whether the team is managed by an identity provider",
+			Computed:            true,
+		},
+		"organization_role": resourceSchema.StringAttribute{
+			MarkdownDescription: "The role assigned to the organization",
+			Optional:            true,
+			Validators: []validator.String{
+				stringvalidator.OneOf(string(iam.ORGANIZATIONOWNER),
+					string(iam.ORGANIZATIONMEMBER),
+					string(iam.ORGANIZATIONBILLINGADMIN),
+				),
+			},
+		},
+		"workspace_roles": resourceSchema.SetNestedAttribute{
+			NestedObject: resourceSchema.NestedAttributeObject{
+				Attributes: ResourceWorkspaceRoleSchemaAttributes(),
+			},
+			Computed:            true,
+			MarkdownDescription: "The roles assigned to the workspaces",
+		},
+		"deployment_roles": resourceSchema.SetNestedAttribute{
+			NestedObject: resourceSchema.NestedAttributeObject{
+				Attributes: ResourceDeploymentRoleSchemaAttributes(),
+			},
+			Computed:            true,
```

Comment:
should prob be optional or maybe even required but can be empty

Reply:
changed to optional - should not be required as teams don't have to have `deployment_roles`